src/provinces/province.cpp

```diff
@@ -1714,8 +1714,9 @@ std::vector<dcon::province_id> make_land_path(sys::state& state, dcon::province_
 
 				if(other_prov.id.index() < state.province_definitions.first_sea_province.index()) { // is land
 					if(has_access_to_province(state, nation_as, other_prov)) {
+						float danger_factor = has_safe_access_to_province(state, nation_as, other_prov) ? 1.f : 4.f;
```

Comment:
this won't make it avoid enemy armies, just provinces that the enemy controls. Once inside enemy territory then (or outside it) it will be just as blind to enemy armies as before

Reply:
made it be wary of any armies not theirs